src/rust-crypto/RoomEncryptor.ts

```diff
@@ -91,7 +103,33 @@ export class RoomEncryptor {
             );
         }
 
+        // Manually call `loadMembersIfNeeded` here, because we want to know if it's the first
+        // time the room is loaded (due to lazy loading), so we can update the tracked users.
+        const fromServer = await this.room.loadMembersIfNeeded();
         const members = await this.room.getEncryptionTargetMembers();
+
+        if (fromServer) {
```

Comment:
Legacy was using another solution (I think more complex), by registering a listener on `RoomStateEvent.Update`. I looked at the code and wasn't conviced that I should do the same.

Instead I call `loadMembersIfNeeded()` to know if it was loaded from server for the first time

Reply:
it seems like, once `loadMembersIfNeeded` has returned true once, it will *always* return true, because it always returns the same promise (`membersPromise`)?

It might be better just to have a boolean in `RoomEncryptor` which is cleared once we call `updateTrackedUsers`.

Reply:
Damn, that's not intuitive. Done as you suggested